### PR TITLE
HLM 5141 hypothesis functional implementation

### PR DESCRIPTION
**Pull request by siddhant-nawale-egov**
Hypothesis Functional implementation

## Summary by CodeRabbit

- **New Features**
    - Introduced a new "Hypothesis Page" for managing campaign assumptions.
    - Added reusable UI components for enhanced user interaction.
    - Implemented a new `ModalWrapper` for customized modal dialogs.
- **Enhancements**
    - Updated styling across various components for improved user experience.
    - Enhanced data fetching and state management in the Upload feature.
    - Improved handling of hypotheses in the employee section with new routes and components.
- **Bug Fixes**
    - Refined validation mechanisms for Excel and GeoJSON uploads with dynamic error messaging.
- **Refactor**
    - Optimized components for better code reuse and maintainability.
- **Style**
    - Upgraded CSS version for the latest styling updates.

### CODE REVIEW

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/Hypothesis.js [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+          RightButtonHandler={closeModal}
+          footerLeftButtonBody={<ButtonType1 text={t("YES")} />}
+          footerRightButtonBody={<ButtonType1 text={t("NO")} />}
+          header={<ModalHeading label={t("HEADING_DELETE_FILE_CONFIRMATION")} />}
+          bodyText={t("HYPOTHESIS_INSTRUCTIONS_DELETE_ENTRY_CONFIRMATION")}
+        />
+      )}
+    </div>
+  );
+};
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/Hypothesis.js:82]: It was already broken down into components like InterractableSection and NonInterractableSection

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/ComonComponents.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  );
+};
+
+// Custom icon component
+export const CustomIcon = (props) => {
+  if (!props.Icon) return null;
+  return <props.Icon fill={props.color} style={{}} {...props} />;
+};
+
+export const ModalHeading = (props) => {
```
Comment (nipunarora-eGov) [inline micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/ComonComponents.js:38]: Pls move common components to react-components library

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Modal.js [hunk 3]
```diff
@@ -105,4 +106,65 @@ const ButtonSelector = (props) => {
   );
 };
 
+
```
Comment (nipunarora-eGov) [inline micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/Modal.js:109]: Modal comp is already there in react-comp library. If we can't use that here then make this comp in comp-lib and name it other than Modal

**Review by nipunarora-eGov — CHANGES_REQUESTED** [review]
Move all the common components under microplan/ to react-comp-library and import from there

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**